src/core/thread/mle_router.cpp

```diff
@@ -1856,6 +1868,15 @@ void MleRouter::HandleStateUpdateTimer(void)
             OT_UNREACHABLE_CODE(break);
         }
 
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+        if (child.IsCslSynchronized() &&
+            TimerMilli::GetNow() - child.GetCslLastHeard() >= Time::SecToMsec(child.GetCslSyncTimeout()))
+        {
+            otLogInfoMle("Child CSL synchronization expired");
+            child.SetCslSynchronized(false);
```

Comment:
We can consider calling `Get<CslTxScheduler>().Update()` here (in case the child was the next in line).